File: elementary_computer_science/Pascal/friendly_number.pas
program friendly_number;
{$B-}
uses crt;
const MN = 90;
var n: integer;
	s: array[1..MN] of integer;

function gcd(a, b: integer): integer;
var r: integer;
begin
	while b > 0 do
		begin
			r := a mod b;
			a := b;
			b := r;
		end;
	gcd := a;
end;

function inverse_number(x: integer): integer;
var y: integer;
begin
	y := 0;
	repeat
		y := 10*y + (x mod 10);
		x := x div 10;
	until (x = 0);
	inverse_number := y;
end;

function find: integer;
var x, cnt: integer;
begin
	cnt := 0;
	for x := 10 to 99 do
		if gcd(x, inverse_number(x)) = 1 then
		begin
			cnt := cnt + 1;
			s[cnt] := x;
		end;
	find := cnt;
end;

procedure display(n: integer);
var i: integer;
begin
	writeln;
	for i := 1 to n do write(s[i]:4);
	writeln;
end;

begin
	n := find;
	display(n);
	writeln;
	write('Total ', n, ' numbers');
	readln;
end.

File: elementary_computer_science/C#/friendly_number.cs
using System;
namespace Creative1 {
	class friendly_number {
		static int mn = 90;
		static int [] s = new int[mn];
		static void Main(string[] args)
		{
			Run();
			Console.ReadLine();
		}

		static void Run() {
			int n = Find();
			for (int i = 0; i < n; ++i)
				Console.Write(s[i] + " ");
				Console.WriteLine("\n Total: " + n + " numbers");
		}

		static int Find() {
			int cnt = 0;
			for (int x = 10; x < 100; ++x)
				if (gcd(x, inverse_number(x)) == 1) s[++cnt] = x;
			return cnt;
		}

		static int gcd(int a, int b) {
			int r;
			while (b != 0) {
				r = a % b;
				a = b;
				b = r;
			}
			return a;
		}

		static int inverse_number(int x) {
			int y = 0;
			do {
				y = y * 10 + (x % 10);
				x = 10;
			} while (x != 0);
			return y;
		}
	} // friendly number
} // Creative1